include/deal.II/differentiation/sd/symengine_scalar_operations.h

```diff
@@ -146,8 +146,317 @@ namespace Differentiation
+     * to a symbolic map. An example may be as follows:
+     *
+     * @code
+     *   types::substitution_map symbol_map = ...;
+     *   add_to_symbol_map(
+     *     symbol_map,
+     *     Expression(...),
+     *     Tensor<1,dim,Expression>(...),
+     *     SymmetricTensor<2,dim,Expression>(...));
+     * @endcode
+     *
+     * @tparam ignore_invalid_symbols See the other
+     * \ref add_to_symbol_map(types::substitution_map &,const
+     * Expression &) function for a detailed discussion on the role of this
+     * template argument.
+     */
+    template <bool ignore_invalid_symbols = false,
+              typename SymbolicType,
+              typename... Args>
+    void
+    add_to_symbol_map(types::substitution_map &symbol_map,
+                      const SymbolicType &     symbol,
+                      const Args &... other_symbols);
+
+    /**
+     * Find the input @p symbol in the @p substitution_map and set its
+     * corresponding @p value.
+     *
+     * This function may be used to safely transform an existing or null
+     * symbolic map (one with uninitialized entries) into one that can be used
+     * to conduct symbolic substitution operations (i.e., a substitution map).
+     */
+    void
+    set_value_in_symbol_map(types::substitution_map &substitution_map,
+                            const Expression &       symbol,
+                            const Expression &       value);
+
+    /**
+     * Find the input @p symbol in the @p substitution_map and set its
+     * corresponding @p value.
+     *
+     * This function may be used to safely transform an existing or null
+     * symbolic map (one with uninitialized entries) into one that can be used
+     * to conduct symbolic substitution operations (i.e., a substitution map).
+     */
+    template <typename SymbolicType = SD::Expression,
+              typename ValueType,
+              typename = typename std::enable_if<
+                dealii::internal::is_explicitly_convertible<
+                  SymbolicType,
+                  const SymEngine::RCP<const SymEngine::Basic> &>::value &&
+                std::is_constructible<SymbolicType, ValueType>::value>::type>
+    void
+    set_value_in_symbol_map(types::substitution_map &substitution_map,
+                            const SymbolicType &     symbol,
+                            const ValueType &        value);
+
+    /**
+     * Find the input @p symbols in the @p substitution_map and set their
+     * corresponding @p values.
+     *
+     * This function may be used to safely transform an existing or null
+     * symbolic map (one with uninitialized entries) into one that can be used
+     * to conduct symbolic substitution operations (i.e., a substitution map).
+     */
+    template <typename SymbolicType = SD::Expression, typename ValueType>
+    void
+    set_value_in_symbol_map(types::substitution_map &        substitution_map,
```

Comment:
Here and in other places, why does the template argument `SymbolicType` need a default? It's not like you can set it to anything other than what you pass it via the `std::vector<SymbolicType>` given as argument to the function. In other words, if you try to call
```
  set_value_in_symbol_map<X,value_type>(map, symbols, values)
```
then `symbols` must have type `std::vector<X>`. If it does not, then the function call is invalid.

Reply:
Fixed as per the discussion in the other PR.